### ISSUE TITLE
Improve AsNoTracking integration tests after EFCore5 upgrade

### ISSUE THREAD

**Issue by Clivar** [ardalis/Specification#73]
Integration tests for EFCore have all entities tracked by default as they are seeded, To make tracking tests work, you are required to manually detach the tested entity first. This is not a perfect solution and may produce side effects.

EFCore5 supplies `ChangeTracker.Clear` for cases like this which is much faster and side-effect free (for example while working with multiple entities or Include's)

Involved PR #72 
Documentation about [ChangeTracker.Clear Method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.changetracking.changetracker.clear?view=efcore-5.0)

**Comment by ardalis** [comment]
Thanks, we'll look at this once we have a package depending on EFCore 5.

**Comment by fiseni** [comment]
Hey @Clivar,

Do you want to update your test now that we upgraded to `EF Core 5`? Your test is [here](https://github.com/ardalis/Specification/blob/3eeeeb66f070643d64570539e7681af15968dd20/ArdalisSpecificationEF/tests/Ardalis.Specification.EF.IntegrationTests/Repository_ListAsync.cs#L179). Please work with `v5` branch. Pull that branch and PR to that branch.

It doesn't hurt to have this kind of integration test, but I don't think it's crucial from the package's perspective. For us, it's important to test that if this feature is used then the flag is saved in the specification, and the evaluator passes the operation to the provider. That's all we do. The actual implementation is done on the provider's side (EF Core), and it's not our concern. On the other side, user's in their code-base might have full integration tests, and test for their scenarios.

**Comment by fiseni** [comment]
If you'll be working on this, please add these unit tests. It's a simple one. Sounds OK?

```c#
public class SpecificationBuilderExtensions_AsNoTracking
{
    [Fact]
    public void DoesNothing_GivenSpecWithoutAsNoTracking()
    {
        var spec = new StoreEmptySpec();

        spec.AsNoTracking.Should().Be(false);
    }

    [Fact]
    public void FlagsAsNoTracking_GivenSpecWithAsNoTracking()
    {
        var spec = new CompanyByIdAsUntrackedSpec(1);

        spec.AsNoTracking.Should().Be(true);
    }
}
```

**Comment by Clivar** [comment]
>  For us, it's important to test that if this feature is used then the flag is saved in the specification, and the evaluator passes the operation to the provider. That's all we do
Basically what I was trying to achieve is exactly that. There seems to be no good way (that I know at least) to test the evaluator for AsNoTracking(). Other tests can rely on `result` to check if it applied correctly. For lack of better option I used efcore to validate it but I do agree this is also basically testing efcore's behavior.

For now I updated my tests to my suggestion which at least gives it a better AAA (Arrange Act Assert) look.
